src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -148,27 +173,44 @@ public static string HtmlDecode(string value)
                 return value;
             }
 
-            // Don't create StringBuilder if we don't have anything to encode
             if (!StringRequiresHtmlDecoding(value))
             {
                 return value;
             }
 
-            StringBuilder sb = StringBuilderCache.Acquire(value.Length);
-            HtmlDecode(value, sb);
-            return StringBuilderCache.GetStringAndRelease(sb);
+            var sb = new ValueStringBuilder(value.Length);
+            HtmlDecode(value, ref sb);
+            return sb.ToString();
         }
 
         public static void HtmlDecode(string value, TextWriter output)
         {
-            output.Write(HtmlDecode(value));
+            if (output == null)
+            {
+                throw new ArgumentNullException(nameof(output));
+            }
+
+            if (string.IsNullOrEmpty(value))
+            {
+                output.Write(value);
+                return;
+            }
+
+            if (!StringRequiresHtmlDecoding(value))
```

Comment:
The code was not previously doing this check. If we have to do it, it might be nice for StringRequiresHtmlDecoding to return the index of the first character to decode, so HtmlDecode can copy up to that point verbatim, the same optimization happening with IndexOfHtmlEncodingChars.
Without that, I am not sure whether this will make it slower or not.

Reply:
It was doing this check before, in the `HtmlDecode(string value)` method (which was being called by this one).